### PR TITLE
Preserve Marshal typing improvements from #1905

### PR DESCRIPTION
**Pull request by erights**
The #1905 spike is idle for now. But I wanted to make sure that these improvements to marshal and same-structure don't get lost.

### CODE REVIEW

File: packages/marshal/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "@agoric/marshal",
   "version": "0.2.7-dev.0",
   "description": "marshal",
-  "main": "marshal.js",
+  "main": "index.js",
```
Comment (michaelfig) [inline packages/marshal/package.json:5]: Please add `"*.js"` to the `"files"` list.

File: packages/marshal/src/marshal.js [hunk 2]
```diff
@@ -445,13 +462,33 @@ function makeReviverIbidTable(cyclePolicy) {
...
+ */
 export function makeMarshal(
-  convertValToSlot = identityFn,
+  convertValToSlot = defaultValToSlotFn,
   convertSlotToVal = defaultSlotToValFn,
 ) {
+  /**
+   * @template Slot
+   * @param {Passable} val
+   * @param {Slot[]} slots
```
Comment (michaelfig) [inline packages/marshal/src/marshal.js:487]: I used to like `Slot[]` better, but found that the JSDoc linter prefers `Array<Slot>`. YMMV.

File: packages/marshal/src/types.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+ * An OnlyData value is PureData when it contains no hidden mutable state,
+ * e.g., when none of its pass-by-copy composite data objects are proxies. This
+ * cannot be determined by inspection. It can only be achieved by trusted
+ * construction. A PureData value cannot be used as a communications channel,
+ * and can therefore be safely shared with subgraphs that should not be able
+ * to communicate with each other.
+ */
+
+/**
+ * @typedef Remotable
```
Comment (michaelfig) [inline packages/marshal/src/types.js]: I think `@typedef` needs a type and a name.

```suggestion
 * @typedef {SOMETHING} Remotable
```

File: packages/same-structure/package.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "name": "@agoric/same-structure",
   "version": "0.0.12-dev.0",
   "description": "Deep equals",
-  "main": "src/sameStructure.js",
+  "main": "index.js",
```
Comment (michaelfig) [inline packages/same-structure/package.json:5]: Please add `"*.js"` to the `"files"` list.

**Review by michaelfig — APPROVED** [review]
Minor changes proposed.  Feel free to merge when CI passes.

File: packages/marshal/package.json (hunk 1 above)
Comment (erights) [reply packages/marshal/package.json:5]: Which "files" list? The only I see in package.json is for ava at
https://github.com/Agoric/agoric-sdk/blob/af0a2b041744a75f634936ab50c1be28ec3dbf32/packages/marshal/package.json#L96-L99
Adding "*.js" there doesn't seem right and doesn't work anyway.

File: packages/marshal/src/marshal.js (hunk 2 above)
Comment (erights) [reply packages/marshal/src/marshal.js:487]: In what way does JSDoc prefer the other?

Other things being equal, I still prefer `Slot[]`

File: packages/marshal/src/types.js (hunk 3 above)
Comment (erights) [reply packages/marshal/src/types.js]: Done

File: packages/same-structure/package.json (hunk 4 above)
Comment (erights) [reply packages/same-structure/package.json:5]: Aha. I see this one. I'll guess that I should just add one like this to marshal/pacjage.json
Done.

File: packages/marshal/package.json (hunk 1 above)
Comment (erights) [reply packages/marshal/package.json:5]: After seeing same-structure/package.json 's "files", I found an explanation at https://docs.npmjs.com/cli/v6/configuring-npm/package-json#files and added one here as
```js
  "files": [
    "src/",
    "*.js",
    "NEWS.md"
  ],
```

**Pull request merged**